viewport/viewport.go

```diff
@@ -292,6 +298,10 @@ func (m Model) updateAsModel(msg tea.Msg) (Model, tea.Cmd) {
 	var cmd tea.Cmd
 
 	switch msg := msg.(type) {
+	case tea.WindowSizeMsg:
+		if m.PastBottom() {
+			m.SetYOffset(m.maxYOffset())
+		}
```

Comment:
This assumes that the width of the viewport is tied to the window width, which will not always be the case: viewports can have a fixed size. Additionally, this requires the user to flow `tea.WindowSizeMsgs` through a viewport's update. It would be better to tie any wrapping to the size.

In order to achieve this we'll probably need to deprecate `Width` and introduce `SetWidth()` and a less-than-ideal `GetWidth()`. In `v2` we can correct this, however.